inference-engine/tests/functional/inference_engine/ngraph_reader/interpolate_tests.cpp

```diff
@@ -220,3 +220,183 @@ TEST_F(NGraphReaderTests, ReadInterpolate2Network) {
+        <edge from-layer="4" from-port="4" to-layer="5" to-port="0"/>
+    </edges>
+</net>
+)V0G0N";
+    std::string modelV5 = R"V0G0N(
+<net name="Network" version="5" precision="FP32" batch="1">
+    <layers>
+        <layer id="0" name="data" precision="FP32" type="Input">
+            <output>
+                <port id="0">
+                    <dim>1</dim>
+                    <dim>2</dim>
+                    <dim>300</dim>
+                    <dim>300</dim>
+                </port>
+            </output>
+        </layer>
+        <layer id="1" name="output_shape" precision="I32" type="Const">
+            <output>
+                <port id="0">
+                    <dim>2</dim>
+                </port>
+            </output>
+            <blobs>
+                <custom offset="0" size="8"/>
+            </blobs>
+        </layer>
+        <layer id="2" name="scales" precision="FP32" type="Const">
+            <output>
+                <port id="0">
+                    <dim>2</dim>
+                </port>
+            </output>
+            <blobs>
+                <custom offset="8" size="8"/>
+            </blobs>
+        </layer>
+        <layer id="3" name="axes" precision="I32" type="Const">
+            <output>
+                <port id="0">
+                    <dim>2</dim>
+                </port>
+            </output>
+            <blobs>
+                <custom offset="16" size="8"/>
+            </blobs>
+        </layer>
+        <layer id="4" name="interpolate" precision="FP32" type="Interpolate">
+            <data antialias="False" coordinate_transformation_mode="asymmetric" cube_coeff="123" mode="nearest" nearest_mode="floor" pads_begin="2,3,4,5" pads_end="6,7,8,9" shape_calculation_mode="sizes"/>
+            <input>
+                <port id="0">
+                    <dim>1</dim>
+                    <dim>2</dim>
+                    <dim>300</dim>
+                    <dim>300</dim>
+                </port>
+                <port id="1">
+                    <dim>2</dim>
+                </port>
+                <port id="2">
+                    <dim>2</dim>
+                </port>
+                <port id="3">
+                    <dim>2</dim>
+                </port>
+            </input>
+            <output>
+                <port id="4" precision="FP32">
+                    <dim>9</dim>
+                    <dim>12</dim>
+                    <dim>600</dim>
+                    <dim>900</dim>
+                </port>
+            </output>
+        </layer>
+    </layers>
+    <edges>
+        <edge from-layer="0" from-port="0" to-layer="4" to-port="0"/>
+        <edge from-layer="1" from-port="0" to-layer="4" to-port="1"/>
+        <edge from-layer="2" from-port="0" to-layer="4" to-port="2"/>
+        <edge from-layer="3" from-port="0" to-layer="4" to-port="3"/>
+    </edges>
+</net>
+)V0G0N";
+    compareIRs(model, modelV5, 24, [](Blob::Ptr& weights) {
```

Comment:
Data from input `scales` are calculated as `scales[i] = sizes[i] / padded_input_shape[axes[i]]`, `i in range(0, len(sizes))`. Here `padded_input_shape = input_shape + pads_begin + pads_end`. Hence, at this test, since `input_shape = [1, 2, 300, 300]`, `pads_begin = [2, 3, 4, 5]`, `pads_end = [6, 7, 8, 9]`, `axes = [2, 3]`, `sizes = [600, 900]`, then `padded_input_shape = [9, 12, 312, 314]`, and `scales = [600.0 / 312.0, 900.0 / 314.0]`, not `[2.0, 2.0]`.

Moreover, output shape for not interpolated axes is equal to `padded_ input_shape`, i.e. output shape of Interpolate-4  from this test is `[9, 12, 600, 900]`.

Reply:
This interpolate have `shape_calculation_mode="sizes"` so the `scales` input does not affect the shape calculation, so it can be arbitrary. Do you agree?

Reply:
I agree, `scales` does not affect the **shape** calculation. But an output shape of Interpolate-4 is incorrect in this test: output shape should be `[9, 12, 600, 900]`.

Reply:
Oh, I see now. The output shape of `Result` operation was wrong (the output shape of `Interpolate-4` was correct). Unit test updated.